templates/workspace_services/guacamole/user_resources/guacamole-azure-windowsvm/template_schema.json

```diff
@@ -62,11 +62,81 @@
         "title": "Shared storage",
         "default": true,
         "description": "Enable access to shared storage"
+      },
+      "enable_shutdown_schedule": {
+        "$id": "#/properties/enable_shutdown_schedule",
+        "type": "boolean",
+        "title": "Enable Shutdown Schedule",
+        "default": false,
+        "description": "Enable automatic shutdown schedule for the VM"
       }
     },
-    "uiSchema": {
-      "admin_username": {
-        "classNames": "tre-hidden"
+    "allOf": [
+      {
+        "if": {
+          "properties": {
+            "enable_shutdown_schedule": {
+              "const": true
+            }
+          },
+          "required": [
+            "enable_shutdown_schedule"
+          ]
+        },
+        "then": {
+          "properties": {
+            "shutdown_time": {
+              "type": "string",
+              "title": "Shutdown Time",
+              "description": "Time of day to shutdown the VM (HHmm format), e.g. 1800",
+              "pattern": "^([01]?[0-9]|2[0-3])[0-5][0-9]$",
+              "default": "1800"
+            },
+            "shutdown_timezone": {
+              "type": "string",
+              "title": "Shutdown Timezone",
+              "description": "Timezone for the shutdown schedule",
+              "default": "UTC",
+              "enum": [
```

Comment:
This is missing some timezones...but is that a big deal? There are quite a few "30 min" timezones missing, and I suspect the biggest one we might care about are the Australia ones.

Reply:
TBH, I've not check what Azure supports, and if it supports timezones other than the ones you've listed. So, it might also be 100% correct according to the Azure API. If so, ignore me.

Reply:
This is already active on Linux VM, believe it works(!).

A link to the full list is here - https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/virtual_machine#timezone-1

Can add named time zones too, but no way we are going to list them all, hence did this, expect people will need to create their own custom templates with a custom list.

Reply:
agreed